### DISCUSSION TITLE
Type awareness

### DISCUSSION THREAD

**Discussion by jbellis** [community/community#8205] [Copilot Conversations]
I'm finding that often CoPilot correctly guesses the logic I want but it often generates code that can't work because it does not conform to the actual API I'm working with.  For example:

![image](https://user-images.githubusercontent.com/42158/143816258-9aefb67d-40af-4e74-a682-9043936f8a28.png)

The signature of approximate is

`public static List<Vector2> approximate(List<Vector2> points, int seconds)`

allSites is a List, so Copilot is suggesting passing (Set, List) to (List, int).

Is addressing this within the scope of Copilot's goals?

**Comment by Awendel** [comment]
second this
type information should definitely be added to the prompt, otherwise it performs worse than intellisense at the moment,
especially for languages that have strong types (such as typescript, ...)
